shared/src/androidMain/kotlin/org/jetbrains/kotlinconf/AndroidNotificationService.kt

```diff
@@ -0,0 +1,190 @@
+package org.jetbrains.kotlinconf
+
+import android.Manifest
+import android.app.AlarmManager
+import android.app.NotificationChannel
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Build
+import androidx.activity.result.ActivityResultLauncher
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.content.ContextCompat
+import androidx.core.content.getSystemService
+import kotlinx.coroutines.channels.Channel
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.toInstant
+
+private const val EXTRA_TITLE = "title"
+private const val EXTRA_MESSAGE = "message"
+private const val EXTRA_NOTIFICATION_ID = "notificationId"
+private const val EXTRA_ICON_ID = "iconId"
+private const val NOTIFICATION_CHANNEL_ID = "channel_all_notifications"
```

Comment:
Should we introduce separate channels for different notification types (session alerts vs news)?

Reply:
Not without putting in a lot more work for managing these channels. The ideal setup for Android users would probably be to use 3 different notification channels and not have in-app settings for notification categories at all, only a button that sends them to the system settings where they can configure each channel. But it's a bunch of extra work just for the one platform, so not critical for the time being.

(We can't read or change the state of notification channels from the app, so we can't sync our own toggles to what the user sets on the channels in settings.)

Reply:
On the bright side, we'll bump the minimum API to 26 on the icons PR, which is where notification channels were introduced, so that eliminates a whole problem where on 24-25 we'd have to have the in-app switcher instead anyway.